### DISCUSSION TITLE
License and copyright info in pygments.css?

### DISCUSSION THREAD

**Discussion by buhtz** [pygments/pygments#2679] [Q&A] [answered]
I would like to know if there is a license for the pygments.css file.

I am upstream maintainer of [Hyperorg](https://codeberg.org/buhtz/hyperorg). The application convert org-files (from Emacs org-mode or orgroam) to HTML. For code blocks it uses `pygments`.

The pygments related CSS is extracted on the fly from the pygments package like this and then stored as `pygments.css`:

    css = pygments.formatters.HtmlFormatter().get_style_defs('pre')

See [original code](https://codeberg.org/buhtz/hyperorg/src/commit/1d7c4d621be56ccf440ee3cbe9493958e2d6884e/src/hyperorg/exporter.py#L242-L252).

For all files in my project I do set the license and copyright information conform to SPDX and REUSE specifications ([example](https://codeberg.org/buhtz/hyperorg/src/commit/1d7c4d621be56ccf440ee3cbe9493958e2d6884e/pyproject.toml#L1-L7)). I would like to do the same for your CSS content.

I do generate my `style.css` on the fly. You can see the license string used as comment on top of that file [here](https://codeberg.org/buhtz/hyperorg/src/commit/1d7c4d621be56ccf440ee3cbe9493958e2d6884e/src/hyperorg/stylecss.py#L28-L34).

Kind
Christian Buhtz

**Comment by jeanas** [comment, marked as answer]
This CSS code is not in the Pygments source code directly, it is generated. Generally speaking, copyright applies to a program but never to its output. Legislations do not provide any way for programmers to impose restrictions on how the output of their code is used, only on how the code itself is used or redistributed. So, you can basically do what you want here (and a license wouldn't make sense for this).

**Reply by buhtz** [reply]
Thanks for your reply. That CSS is not taken from a string constant like I do in my "stylecss.py" file?

**Reply by jeanas** [reply]
No, it's generated from the token styles in `pygments/styles/default.py` (by default) by `HtmlFormatter.get_style_defs` (see the code in `pygments/formatters/html.py` like 508).
